.devcontainer/docker-compose-devcontainer.yml

```diff
@@ -0,0 +1,194 @@
+services:
+  devcontainer:
+    image: 'mcr.microsoft.com/devcontainers/typescript-node:1-18-bullseye'
+    volumes:
+      - ..:/workspaces:cached # mounts our local directory (git root) to the container in /workspaces directory (with access mode of cached)
+
+      # We need to override all node_modules directories
+      # by mounting an empty volume to avoid conflicting
+      # CPU build instructions between the host machine and the DevContainer's docker image
+      # (this allows the devcontainer to be based on linux yet work on Apple Silicon etc..)
+      # If you add a new package with a new `node_modules`, it needs to be added here
+      - type: volume
```

Comment:
this ain't gonna be an exhaustive list, there can be any number of further node_modules folders down in deeper indirect dependency folders